### DISCUSSION TITLE
q-date options and limiting options from selected date range

### DISCUSSION THREAD

**Discussion by L-S-ONG** [quasarframework/quasar#12632] [General - Components / Directives / etc]
Hi, 

I am trying to use a range selection and at the same time, I am disabling "future" dates (dates > today).

However, I would like to add another condition, that is, when user select the range start date, I would like to disable (grey out) the date selection from selected date to 90 days _(from the start of selected date range)_  

> For example If I select 23 August 2021, then the calendar date range will only allow me to pick a maximum date range of 90 days from 23 August 2021, that is 21 November 2021. Any dates that comes after 21 November 2021 will be grey out, at the same time, any dates that is after Today's date will be grey out as well

Has anyone tried this before and please let me know if there's any solutions out there

```
<q-date
  v-model="dateRange"
  range
  minimal
  :options="optionsFn"
  @range-start="setModelValueRangeFromDay"
/>
```

```
optionsFn (date) {
  var todayDate = new Date()
  var todayDateRange = moment(todayDate).format('YYYY/MM/DD')
  return date <= todayDateRange
}
```

```
  setModelValueRangeFromDay (from) {
  }
```

**Comment by MilosPaunovic** [comment]
I think you're looking for something in the lines of this:
[https://codepen.io/milospaunovic/pen/abVQrrP?editors=1010](https://codepen.io/milospaunovic/pen/abVQrrP?editors=1010)

**Reply by L-S-ONG** [reply]
Hi,

Thank you for the suggested answer, unfortunately the solution you had does not work in this case. I would need to check for the range start date (you could see from the @range-start) when user selects the starting date range from the calendar and call the **optionsFn (date)** method again.

**Comment by MilosPaunovic** [comment]
If I understand correctly, what you need is a range date selection with available dates from selected range start day and the one 90 days later. If that is the case, Codepen below should solve your problem (mind that it was written in a bit of a hurry, so it probably could be optimized).

[https://codepen.io/milospaunovic/pen/abVQrrP?editors=1011](https://codepen.io/milospaunovic/pen/abVQrrP?editors=1011)
